src/ovirtapi/types.js

```diff
@@ -24,7 +24,8 @@ export type ApiVmStatisticType = {
 }
 
 export type StatisticValueType = {
-  datum: number | Array<number>,
+  firstDatum: number | void,
+  datum: Array<number>,
```

Comment:
`number` isn't the only thing the datum can be ... they can be number or string or objects even.  So probably better to be at least a `number | string | Object`.

Reply:
Do you have any example for string or object, from what I tested is always numbers, undefined is for empty array...

Reply:
@bamsalem just check the code on backend: https://github.com/oVirt/ovirt-engine/blob/9059df4d26f5d327c87495ee9e19cb1876ad92b7/backend/manager/modules/restapi/types/src/main/java/org/ovirt/engine/api/restapi/utils/StatisticResourceUtils.java#L36-L74

BigDecimal, Long and of couse String should be supported as well.

Reply:
> Do you have any example for string or object, from what I tested is always numbers, undefined is for empty array...

Currently, a datum will be a String only for the "disks.usage" statistic.  You need to have a running VM with a valid guest agent running to get that value.  A current install of Fedora or SilverBlue should do the trick.

See: [VmStatisticalQuery.java](https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/VmStatisticalQuery.java) for the full list of VM stats currently generated.